### PR TITLE
Philip/ussd locale files

### PR DESCRIPTION
**Pull request by mango-habanero**
Refactors file fetching to get files from gitlab

### CODE REVIEW

**Review by nolash — COMMENTED** [review]
Please don't:

* remove s3 test
* remove code documentation

File: config_files/public/local_config.ini [hunk 1]
```diff
@@ -25,6 +25,10 @@ self_service_wallet_initial_disbursement     = 5000
 
 [SYSTEM]
 
+[GITLAB]
+base_url = https://gitlab.com/api/v4/projects/
```
Comment (nolash) [inline config_files/public/local_config.ini]: please lets decompose in 

* scheme
* host
* path

File: config.py [hunk 2]
```diff
@@ -414,3 +414,12 @@ def address_from_private_key(private_key):
         TRANSFER_LIMITS[k.upper()] = v 
 except KeyError:
     pass
+
+
+# gitlab configs
+GITLAB_PRIVATE_TOKEN = secrets_parser['GITLAB'].get('private_token')
```
Comment (nolash) [inline config.py]: what's private token? and why do you need all this metadata?

File: files/gitlab.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# standard imports
+import base64
+import logging
+import requests
+
+# platform imports
+import config
+from files.sync import FileSyncer
+
+github_files_logger = logging.getLogger(__name__)
```
Comment (nolash) [inline files/gitlab.py]: let's just always call this "logg"

File: files/gitlab.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def get_files(filename):
+    from urllib.parse import quote
+    content, last_commit_id = None, None
+    url_encoded_filename = quote(filename, safe='').replace('.', '%2E')
```
Comment (nolash) [inline files/gitlab.py]: can we do without this hack please?

File: files/sync.py [hunk 5]
```diff
@@ -5,83 +5,42 @@
...
-class FileSyncer:
-    """Parent class to be overloaded by component providing access to
-    external file resources.
 
-    A function is passed to the constructor by the implementer, which _must_ 
-    fulfull the following interface:
-        * accept a single string as argument for the filename to retrieve
-        * return an object implementing a read(blocksize=int) method
+def source_newer(last_commit_id):
+    if last_commit_id != config.GITLAB_LAST_COMMIT_ID:
```
Comment (nolash) [inline files/sync.py]: well that doesn't necessarily mean newer, now does it? ;)

File: files/sync.py [hunk 6]
```diff
@@ -5,83 +5,42 @@
...
 
-class FileSyncer:
-    """Parent class to be overloaded by component providing access to
-    external file resources.
 
-    A function is passed to the constructor by the implementer, which _must_ 
-    fulfull the following interface:
-        * accept a single string as argument for the filename to retrieve
-        * return an object implementing a read(blocksize=int) method
+def source_newer(last_commit_id):
```
Comment (nolash) [inline files/sync.py]: why renamed. now you are breaking the interface and s3 implementation will not work anymore.

File: test/unit/files/test_files.py [hunk 7]
```diff
@@ -20,12 +20,7 @@ def test_files():
 def test_s3():
     uu = str(uuid.uuid4())
     randomfilename = '/tmp/' + uu + '/foo'
-    fs = S3(
-           config.TEST_BUCKET,
-           randomfilename,
-           key=config.AWS_SES_KEY_ID,
-           secret=config.AWS_SES_SECRET
-           )
+    fs = Gitlab(config.TEST_BUCKET)
```
Comment (nolash) [inline test/unit/files/test_files.py]: why "test bucket" to gitlab?

**Review by nolash — CHANGES_REQUESTED** [review]

**Comment by nolash** [comment]
@PhilipWafula can you resubmit to gitlab please?

File: config.py (hunk 2 above)
Comment (mango-habanero) [reply config.py]: Its a token GitLab uses to authenticate requests to private repositories over the repository files API

File: files/gitlab.py (hunk 4 above)
Comment (mango-habanero) [reply files/gitlab.py]: Unfortunately the API requires that the file path be URL encoded fully and the `urllib.parse` doesn't seem to encode periods.

File: files/sync.py (hunk 5 above)
Comment (mango-habanero) [reply files/sync.py]: It implies different though no ? The idea here being that if a change is made to the files repo then it is worth fetching a new.

File: config.py (hunk 2 above)
Comment (nolash) [reply config.py]: but why does it need to be private? these yml files aren't secret...

File: files/sync.py (hunk 6 above)
Comment (mango-habanero) [reply files/sync.py]: This is my brain at 6 AM having not slept all night, I have reverted the file syncer for now, then I will try and overload it as you intended in its design.

**Comment by mango-habanero** [comment]
Resolved by: https://gitlab.com/grassrootseconomics/sempo-fork/-/merge_requests/7

**Pull request closed without merging**